storage/innobase/handler/ha_innodb.cc

```diff
@@ -23996,3 +24009,62 @@ innodb_encrypt_tables_validate(
 
 	return 0;
 }
+
+/** Validates the possible innodb_redo_log_encrypt_values.
+@param[in]	thd	current session
+@param[in]	var	the system variable innodb_support_xa
+@param[in,out]	var_ptr	the contents of the variable
+@param[in]	save	the to-be-updated value */
+static
+int
+innodb_redo_log_encrypt_validate(
+	THD *		     	thd,
+	struct st_mysql_sys_var *var,
+	void *		     	save,
+	struct st_mysql_value   *value)
+{
+	const char	*redo_log_encrypt_input;
+	char		buff[STRING_BUFFER_USUAL_SIZE];
+	int		len = sizeof(buff);
+
+	ut_a(save != NULL);
+	ut_a(value != NULL);
+
+	redo_log_encrypt_input = value->val_str(value, buff, &len);
+
+	bool legit_value = false;
+	uint use = 0;
+	for (; use < array_elements(redo_log_encrypt_names); use++) {
+		if (innobase_strcasecmp(redo_log_encrypt_input,
+					redo_log_encrypt_names[use]) == 0) {
+			legit_value = true;
+			break;
+		}
+	}
+
+	if (innobase_strcasecmp(redo_log_encrypt_input, "0") == 0) {
+		use = 0;
+		legit_value = true;
+	}
+
+	if (innobase_strcasecmp(redo_log_encrypt_input, "false") == 0) {
+		use = 0;
+		legit_value = true;
+	}
+
+	if (innobase_strcasecmp(redo_log_encrypt_input, "1") == 0) {
+		use = 1;
+		legit_value = true;
+	}
+
+	if (innobase_strcasecmp(redo_log_encrypt_input, "true") == 0) {
+		use = 1;
+		legit_value = true;
+	}
+
+	if (legit_value == false)
```

Comment:
(Github is crashing, this is my full review)
```suggestion
	if (!legit_value)
```